src/main/java/jenkins/branch/BranchProjectFactory.java

```diff
@@ -198,8 +198,8 @@ public P decorate(P project) {
                 if (decorator != null) {
                     // if Project then we can feed the publishers and build wrappers
                     if (project instanceof Project) {
-                        DescribableList<Publisher, Descriptor<Publisher>> publishersList = project.getPublishersList();
-                        DescribableList buildWrappersList = Project.class.cast(project).getBuildWrappersList();
+                        DescribableList<Publisher, Descriptor<Publisher>> publishersList = ((Project) project).getPublishersList();
+                        DescribableList buildWrappersList = ((Project) project).getBuildWrappersList();
```

Comment:
@varmenise We only know it's `P extends Job` from the generic types, so it's not necessarily a `Project`. However we passed the `if` condition so the cast is safe.